pkg/generators/system/config/options.go

```diff
@@ -1,191 +1,84 @@
-		BackendInternalAPIUser:     &pod.SecretValue{Value: spec.Config.Backend.InternalAPIUser},
-		BackendInternalAPIPassword: &pod.SecretValue{Value: spec.Config.Backend.InternalAPIPassword},
-
-		AssetsAWSAccessKeyID:     &pod.SecretValue{Value: spec.Config.Assets.AccessKey},
-		AssetsAWSSecretAccessKey: &pod.SecretValue{Value: spec.Config.Assets.SecretKey},
-		AssetsAWSBucket:          &pod.ClearTextValue{Value: spec.Config.Assets.Bucket},
-		AssetsAWSRegion:          &pod.ClearTextValue{Value: spec.Config.Assets.Region},
-
-		AppSecretKeyBase:                 &pod.SecretValue{Value: spec.Config.SecretKeyBase},
-		SegmentDeletionToken:             &pod.SecretValue{Value: spec.Config.Segment.DeletionToken},
-		SegmentDeletionWorkspace:         &pod.ClearTextValue{Value: spec.Config.Segment.DeletionWorkspace},
-		SegmentWriteKey:                  &pod.SecretValue{Value: spec.Config.Segment.WriteKey},
-		GithubClientID:                   &pod.SecretValue{Value: spec.Config.Github.ClientID},
-		GithubClientSecret:               &pod.SecretValue{Value: spec.Config.Github.ClientSecret},
-		RedHatCustomerPortalClientID:     &pod.SecretValue{Value: spec.Config.RedHatCustomerPortal.ClientID},
-		RedHatCustomerPortalClientSecret: &pod.SecretValue{Value: spec.Config.RedHatCustomerPortal.ClientSecret},
-		DatabaseSecret:                   &pod.SecretValue{Value: spec.Config.DatabaseSecret},
-	}
-
-	if spec.Config.Bugsnag.Enabled() {
-		opts.BugsnagAPIKey = &pod.SecretValue{Value: spec.Config.Bugsnag.APIKey}
-
-		if spec.Config.Bugsnag.ReleaseStage != nil {
-			opts.BugsnagReleaseStage = &pod.ClearTextValue{Value: *spec.Config.Bugsnag.ReleaseStage}
-		}
-
-	} else {
-		opts.BugsnagAPIKey = &pod.SecretValue{Value: saasv1alpha1.SecretReference{Override: util.Pointer("")}}
-	}
-
-	if spec.Config.Assets.Host == nil {
-		opts.AssetsHost = &pod.ClearTextValue{Value: ""}
-	} else {
-		opts.AssetsHost = &pod.ClearTextValue{Value: *spec.Config.Assets.Host}
-	}
-
-	if spec.Config.AccessCode != nil {
-		opts.AccessCode = &pod.SecretValue{Value: *spec.Config.AccessCode}
-	}
-
-	if spec.Config.RedHatCustomerPortal.Realm != nil {
-		opts.RedHatCustomerPortalRealm = &pod.ClearTextValue{Value: *spec.Config.RedHatCustomerPortal.Realm}
-	}
-
-	if spec.Config.SMTP.STARTTLS != nil {
-		opts.SMTPSTARTTLS = &pod.ClearTextValue{Value: fmt.Sprintf("%t", *spec.Config.SMTP.STARTTLS)}
-	}
-
-	if spec.Config.SMTP.STARTTLSAuto != nil {
-		opts.SMTPSTARTTLSAuto = &pod.ClearTextValue{Value: fmt.Sprintf("%t", *spec.Config.SMTP.STARTTLSAuto)}
-	}
-
-	if spec.Config.Zync != nil {
-		opts.ZyncEndpoint = &pod.ClearTextValue{Value: spec.Config.Zync.Endpoint}
-		opts.ZyncAuthenticationToken = &pod.SecretValue{Value: spec.Config.Zync.AuthToken}
-	} else {
-		opts.ZyncAuthenticationToken = &pod.SecretValue{Value: *spec.Config.ZyncAuthToken}
-	}
+// cat <file> | grep secret: | sed -r 's/.*`env:"(\w*)"(\s*secret:"([0-9a-zA-Z\-]*)")?`/Unpack().IntoEnvvar("\1").AsSecretRef("\3"),/gm'
```

Comment:
needed comment?

Reply:
haha, definitely not. It's what I used to migrate from the old syntax to the new one for defining options. Fixed.